test/e2e/issue351_test.go

```diff
@@ -0,0 +1,161 @@
+			ResourceDeletionMode:   "Delete",
+			ServiceAccountRef:      "eunomia-operator",
+		},
+	}
+
+	err = framework.Global.Client.Create(ctx, gitops, &framework.CleanupOptions{TestContext: ctx.TestCtx, Timeout: timeout, RetryInterval: retryInterval})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// When the initial job is created, we will use it as a template to create two additional jobs at the same time
+	err = wait.Poll(retryInterval, 60*time.Second, func() (done bool, err error) {
+		jobs, err := framework.Global.KubeClient.BatchV1().Jobs(ctx.namespace).List(metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+		switch {
+		case len(jobs.Items) == 0:
+			t.Logf("Waiting for job to be created")
+			return false, nil
+		default:
+			t.Logf("Job has been created")
+			return true, nil
+		}
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	gitopsJob, err := GetJob(ctx.namespace, "gitopsconfig-gitops-issue351-", framework.Global.KubeClient)
+
+	firstJob := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "GitOpsConfig",
+			APIVersion: "eunomia.kohls.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "first-job",
+			Namespace: ctx.namespace,
+		},
+		Spec: gitopsJob.Spec,
+	}
+	// The deep copy of the job keeps the selector and selector label that has to be generated by k8s.
+	// Trying to create a job with those set will fail.
+	firstJob.Spec.Template.SetLabels(map[string]string{})
+	firstJob.Spec.Selector.Reset()
+
+	err = framework.Global.Client.Create(ctx, firstJob, &framework.CleanupOptions{TestContext: ctx.TestCtx, Timeout: timeout, RetryInterval: retryInterval})
+	if err != nil {
+		t.Error(err)
+	}
+
+	secondJob := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "GitOpsConfig",
+			APIVersion: "eunomia.kohls.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "second-job",
+			Namespace: ctx.namespace,
+		},
+		Spec: gitopsJob.Spec,
+	}
+	secondJob.Spec.Template.SetLabels(map[string]string{})
+	secondJob.Spec.Selector.Reset()
+
+	err = framework.Global.Client.Create(ctx, secondJob, &framework.CleanupOptions{TestContext: ctx.TestCtx, Timeout: timeout, RetryInterval: retryInterval})
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Wait to make sure both of the jobs finish running
+	err = wait.Poll(retryInterval, 60*time.Second, func() (done bool, err error) {
+		jobOne, _ := GetJob(ctx.namespace, "first-job", framework.Global.KubeClient)
+		jobTwo, _ := GetJob(ctx.namespace, "second-job", framework.Global.KubeClient)
+
+		switch {
+		case jobOne.Status.Succeeded == 1 && jobTwo.Status.Succeeded == 1:
+			t.Logf("Both jobs are done")
+			return true, nil
+		default:
+			t.Logf("Both jobs are not done")
+			return false, nil
+		}
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Ensure the pod is pod is not deleted when the jobs complete
+	err = WaitForPodWithImage(t, framework.Global, ctx.namespace, "hello-world-modes", "hello-app:1.0", retryInterval, timeout)
```

Comment:
`hello-world-modes` ?

Reply:
I updated the test to use a more generic set of test data, the `hello-world-modes` was a result of using the `modes_test.go` as a template for generating this test. I also updated the job to look for the deployment rather than the pod because that's more important from this tests perspective.